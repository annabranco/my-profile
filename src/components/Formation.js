import React from 'react';
import '../styles/components/Formation.css';

class Formation extends React.Component {

	render () {

		const text = this.props.texts.Formation;
		const language = this.props.language;

		return (

			<section className="section__formation">
				<div className="formation__outer">
					<div className="formation__inner">
						<div className="formation__inner--year">1999 - 2005</div>
						<div className="formation__inner--year">2007 - 2011</div>

						<div className="formation__inner--year">2012 - 2013</div>

						<div className="formation__inner--year">2018</div>

					</div>
					<div className="formation__inner">
						<div className="formation__verticalBar"></div>
					</div>
					<div className="formation__inner">
						<div className="formation__inner--horizontalBar">
							<div className="formation__horizontarBar-psy"></div>
							<h2 className="formation__title formation__title-psy">{text[language].psy}</h2>
							<p className="formation__details formation__details-psy">Universidade Gama Filho
								<img className="formation__flag" src="https://www.countryflags.io/br/flat/16.png" alt={text[language].brazil} title={text[language].brazil} />
							</p>
							<p className="formation__details formation__details-psy">{text[language].grade}8,46</p>
						</div>
						<div className="formation__inner--horizontalBar">
							<div className="formation__horizontarBar-ir"></div>
							<h2 className="formation__title formation__title-ir">{text[language].ir}</h2>
							<p className="formation__details formation__details-ir">Universidade Estácio de Sá
								<img className="formation__flag" src="https://www.countryflags.io/br/flat/16.png" alt={text[language].brazil} title={text[language].brazil}  />
							</p>
							<p className="formation__details formation__details-ir">{text[language].grade}9,15</p>

						</div>
						<div className="formation__inner--horizontalBar">
							<div className="formation__horizontarBar-master"></div>
							<h2 className="formation__title formation__title-master">{text[language].master}</h2>
							<p className="formation__details formation__details-master">Universitat Autònoma de Barcelona <img className="formation__flag" src="https://www.countryflags.io/es/flat/16.png" alt={text[language].spain} title={text[language].spain}  />
							</p>
							<p className="formation__details formation__details-master">{text[language].grade}8,00</p>

						</div>
						<div className="formation__inner--horizontalBar">
							<div className="formation__horizontarBar-adalab"></div>
							<h2 className="formation__title formation__title-adalab">Adalab</h2>
						</div>
					</div>

				</div>
			</section>


		);
	}

}

export default Formation;
