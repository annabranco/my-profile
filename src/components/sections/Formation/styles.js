import styled, { css, keyframes } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  fontSubtitle,
  colorYellowBright,
  colorBlueWater,
  colorYellowDark,
  fontTitleAlt
} from '../../../styles/theme';
import { Title } from '../Experiences/styles';

export const Details = styled.p`
  margin: 0 20px;
  font-family: ${({ subtext }) => (subtext ? fontTitleAlt : fontSubtitle)};
  font-size: 0.9rem;
  line-height: 1;

  @media all and (min-width: 768px) {
    margin: 0 40px;
    font-size: 1rem;
    line-height: 1.3;
  }
  &:first-of-type {
    margin-top: 5px;
  }
`;
Details.displayName = 'Details';

const goIn = (formationItems, index) => keyframes`
  0% {
    margin-left : -100vw;
  }

  100% {
      margin-left: ${`${
        (10 / formationItems) * ((index + 1) * formationItems)
      }px`};  }
`;

export const DetailsArea = styled.div`
  margin-top: 10px;
  width: 80%;

  ${({ formationItems, index }) =>
    formationItems &&
    css`
      animation: ${goIn(formationItems, index)} 2s forwards;
    `}

  @media all and (min-width: 768px) {
    margin-top: 0;

    ${({ formationItems, index }) =>
      formationItems &&
      css`
        margin-left: ${`${
          (60 / formationItems) * ((index + 1) * formationItems)
        }px`};
      `}
  }
`;
DetailsArea.displayName = 'Details Area';

export const FakeText = styled.p`
  width: 100%;
  font-size: 0.9rem;
  line-height: 0.8;
  text-align: center;
`;
FakeText.displayName = 'Fake Text';

export const FormationArea = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 0 auto;
  height: 100%;
  width: 100%;
`;
FormationArea.displayName = 'Formation Area';

export const FormationItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 20px 0;
  height: auto;
  width: 100%;
`;
FormationItem.displayName = 'Formation Item';

export const FormationTitle = styled(Title)`
  &:hover {
    letter-spacing: 0.1rem;
    color: ${colorYellowDark};
  }
`;
FormationTitle.displayName = 'Formation Title';

export const FormationWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-end;
  margin: 50px auto 0;
  padding-right: 5vw;
  height: ${`${window.innerHeight}px`};
  width: 100%;

  @media all and (min-width: 768px) {
    margin: 0 auto;
    height: auto;
    justify-content: space-around;
  }
`;
FormationWrapper.displayName = 'Formation Wrapper';

export const SectionFormation = styled.section`
  z-index: 6;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 0.95;
      transform: none;
      height: 100%;
      width: 100vw;

      &:hover {
        border: 2px solid ${colorBlack};
        opacity: 0.95;
      }
    `}

  @media all and (min-width: 768px) {
    position: absolute;
    bottom: 50px;
    left: 44%;
    opacity: 0.5;
    transform: perspective(5em) rotateX(50deg);
    outline: none;
    border: 1px solid ${colorBlack};
    border-radius: 10px 0;
    height: 40px;
    width: 60px;
    overflow: hidden;
    cursor: pointer;
    box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
      inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};
    background-image: linear-gradient(
      to top right,
      ${colorWhite},
      ${colorBlueWater},
      ${colorBlueLight}
    );

    &:hover {
      opacity: 0.7;
      border: 3px solid ${colorYellowBright};
    }

    ${({ visible }) =>
      visible &&
      css`
        top: 40px;
        left: 20vw;
        opacity: 0.95;
        transform: none;
        border: 2px solid ${colorBlack};
        border-radius: 40px 0;
        height: 90vh;
        width: 60vw;
        padding: 15px;
        cursor: default;

        &:hover {
          border: 2px solid ${colorBlack};
          opacity: 0.95;
        }
      `}
  }
`;
SectionFormation.displayName = 'FORMATION Section';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = 'Vertical Bar';
