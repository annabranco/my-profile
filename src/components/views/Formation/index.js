import React from 'react';
import { PropTypes } from 'prop-types';
import {
  experiencesTextPropType,
  seabedElementsPropType,
  globalTextsPropType,
  formationPropType
} from '../../../types';
import { isDesktop } from '../../../utils/device';
// import {  } from './styles.js';
import {
  SeabedCloseButton,
  TextFindSomething
} from '../../containers/Seabed/styles';
import {
  FakeText,
  SectionFormation,
  FormationWrapper,
  FormationInnerWrapper,
  FormationYear,
  FormationVerticalBar,
  FormationSeparator,
  FormationHorizontalBar,
  FormationTitle,
  FormationDetails,
  FormationFlag,
  FormationMore,
  FormationMoreLink
} from './styles';

const Formation = ({
  texts,
  globalTexts,
  formation,
  status: { read, visible },
  onClickOpen,
  onClickClose
}) => (
  <>
    {!read && isDesktop ? (
      <>
        <TextFindSomething>{texts.find}</TextFindSomething>
        <TextFindSomething>{texts.investigate}</TextFindSomething>
      </>
    ) : null}
    {/* <div
        className={`section__experiences ${visible && SHOW_ACTION}`}
        onClick={() => onClickOpen('experiences')}
        role="button"
        aria-label={globalTexts.open}
        tabIndex={0}
      > */}
    <SectionFormation
      onClick={() => onClickOpen('experiences')}
      role="button"
      aria-label={globalTexts.open}
      tabIndex={0}
      visible={visible}
    >
      {!visible ? (
        <>
          <FakeText>- - --- - --</FakeText>
          <FakeText>- -- -- - --</FakeText>
          <FakeText>- -- ---- --</FakeText>
          <FakeText>- - - --- --</FakeText>
        </>
      ) : (
        <FormationWrapper>
          <SeabedCloseButton
            onClick={() => onClickClose('experiences')}
            type="button"
            aria-label={globalTexts.close}
          >
            X
          </SeabedCloseButton>

          <FormationInnerWrapper>
            <FormationYear>09/2017 - 05/2018</FormationYear>
            <FormationYear>10/2015 - 11/2015</FormationYear>
            <FormationYear>05/2005 - 12/2014</FormationYear>
          </FormationInnerWrapper>
          <FormationInnerWrapper>
            <FormationVerticalBar />
          </FormationInnerWrapper>
          <FormationInnerWrapper>
            <FormationSeparator>
              <FormationHorizontalBar />
              <FormationTitle>{texts.ict}</FormationTitle>
              <FormationDetails>
                Servicios Profesionales Sociales, Madrid.
                <FormationFlag
                  src="https://www.countryflags.io/es/flat/16.png"
                  alt={texts.spain}
                  title={texts.spain}
                />
              </FormationDetails>
              <FormationDetails>{texts.ictDetails}</FormationDetails>
            </FormationSeparator>
            <FormationSeparator>
              <FormationHorizontalBar />
              <FormationTitle>{texts.eru}</FormationTitle>
              <FormationDetails>
                Cruz Roja Española
                <FormationFlag
                  src="https://www.countryflags.io/es/flat/16.png"
                  alt={texts.spain}
                  title={texts.spain}
                />
              </FormationDetails>
              <FormationDetails>{texts.eruDetails}</FormationDetails>
            </FormationSeparator>
            <FormationSeparator>
              <FormationHorizontalBar />
              <FormationTitle>{texts.tj}</FormationTitle>
              <FormationDetails>
                Tribunal de Justiça do Estado do Rio de Janeiro{' '}
                <FormationFlag
                  src="https://www.countryflags.io/br/flat/16.png"
                  alt={texts.brazil}
                  title={texts.brazil}
                />
              </FormationDetails>
              <FormationDetails>{texts.tjDetails}</FormationDetails>
            </FormationSeparator>
          </FormationInnerWrapper>
          <FormationMore>
            {texts.linkedin}
            <FormationMoreLink
              href="https://www.linkedin.com/in/annabranco/"
              target="_Blank"
              rel="noopener noreferrer"
            >
              <i className="fab fa-linkedin-in icon-linkedin" />
            </FormationMoreLink>
          </FormationMore>
        </FormationWrapper>
      )}
    </SectionFormation>
  </>
);

Formation.propTypes = {
  texts: experiencesTextPropType.isRequired,
  globalTexts: globalTextsPropType.isRequired,
  formation: PropTypes.arrayOf(formationPropType).isRequired,
  status: seabedElementsPropType.isRequired,
  onClickOpen: PropTypes.func.isRequired,
  onClickClose: PropTypes.func.isRequired
};

export default Formation;
