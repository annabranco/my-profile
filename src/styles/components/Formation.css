.section__formation {
  height: 100vh;
  position: relative;
  display: fixed;
  background-image: linear-gradient(rgba(230, 239, 251, 0.98), rgba(230, 239, 251, 0.93), rgba(189, 214, 246, 0.98)), url("https://data.whicdn.com/images/303133698/original.gif");
  background-position: center; }

.formation__outer {
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
  margin: 0 auto;
  height: auto;
  width: 80vw; }

.formation__inner {
  height: auto; }

.formation__inner--year {
  margin-right: 15px;
  width: 40px;
  text-align: center;
  font-family: "Alegreya Sans SC", sans-serif;
  font-size: 1.2em;
  color: #4b0a62; }

.formation__verticalBar {
  border-right: 5px solid #2b2f59;
  height: 90vh; }

.formation__inner--horizontalBar,
.formation__inner--year {
  height: 130px; }

.formation__horizontarBar-adalab,
.formation__horizontarBar-ir,
.formation__horizontarBar-master,
.formation__horizontarBar-psy {
  display: inline-block;
  border-top: 5px solid #2b2f59; }

.formation__horizontarBar-psy {
  width: 20px; }
  @media all and (min-width: 768px) {
    .formation__horizontarBar-psy {
      width: 0;
      transition: width ease 2s; }
      .formation__horizontarBar-psy.comeIn {
        width: 15vw; } }

.formation__horizontarBar-ir {
  width: 20px; }
  @media all and (min-width: 768px) {
    .formation__horizontarBar-ir {
      width: 0;
      transition: width ease 2s; }
      .formation__horizontarBar-ir.comeIn {
        width: 30vw; } }

.formation__horizontarBar-master {
  width: 20px; }
  @media all and (min-width: 768px) {
    .formation__horizontarBar-master {
      width: 0;
      transition: width ease 2s; }
      .formation__horizontarBar-master.comeIn {
        width: 45vw; } }

.formation__horizontarBar-adalab {
  width: 20px; }
  @media all and (min-width: 768px) {
    .formation__horizontarBar-adalab {
      width: 0;
      transition: width ease 2s; }
      .formation__horizontarBar-adalab.comeIn {
        width: 60vw; } }

.formation__title {
  margin-left: 20px;
  display: inline;
  font-family: "Acme", sans-serif;
  font-size: 1.4em;
  color: #3e76c1; }
  .formation__title:hover {
    letter-spacing: 0.1rem;
    color: #4681d0; }

@media all and (min-width: 768px) {
  .formation__title-adalab,
  .formation__title-ir,
  .formation__title-master,
  .formation__title-psy {
    opacity: 0;
    transition: opacity ease 2s; }
    .formation__title-adalab.comeIn,
    .formation__title-ir.comeIn,
    .formation__title-master.comeIn,
    .formation__title-psy.comeIn {
      opacity: 1; } }

.formation__details {
  margin-left: 10px;
  font-family: "Corben", cursive;
  font-size: 1rem; }

@media all and (min-width: 768px) {
  .formation__details-ir,
  .formation__details-master,
  .formation__details-psy {
    opacity: 0;
    transition: opacity ease 2s; }
    .formation__details-ir.comeIn,
    .formation__details-master.comeIn,
    .formation__details-psy.comeIn {
      opacity: 1; } }

.formation__flag {
  margin-left: 5px;
  vertical-align: middle; }
