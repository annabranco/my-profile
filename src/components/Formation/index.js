import React from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import { isDesktop } from '../../utils/device';
import { getFlagURL } from '../../utils/icons';
import {
  Details,
  DetailsArea,
  FakeText,
  FormationArea,
  FormationItem,
  FormationTitle,
  FormationWrapper,
  SectionFormation,
  VerticalBar
} from './styles';
import { SeabedCloseButton, TextFindSomething } from '../Seabed/styles';
import { HorizontalBar } from '../elements/HorizontalBar/styles';
import { CountryFlag, DateArea, TextDate } from '../Experiences/styles';
import { Title } from '../ScrollSection/styles';
import { seabedElementsPropType } from '../../types';
import {
  formationSelector,
  globalTextsSelector,
  currentLanguageSelector,
  seabedTextsSelector
} from '../../redux/selectors';

const FORMATION_ON_TOP = 'older'; // newer or older

const Formation = ({
  onClickClose,
  onClickOpen,
  status: { read, visible }
}) => {
  const formation = useSelector(formationSelector);
  const languageSelected = useSelector(currentLanguageSelector);
  const texts = useSelector(seabedTextsSelector);
  const globalTexts = useSelector(globalTextsSelector);

  const customOrder = (x, y) => {
    if (FORMATION_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <>
      {!read && isDesktop ? (
        <>
          <TextFindSomething>{texts.find}</TextFindSomething>
          <TextFindSomething>{texts.investigate}</TextFindSomething>
        </>
      ) : null}
      <SectionFormation
        id="Formation Section"
        onClick={() => onClickOpen('formationSection')}
        role="button"
        aria-label={globalTexts.open}
        tabIndex={0}
        visible={visible}
      >
        {!visible ? (
          <>
            <FakeText>- - --- - --</FakeText>
            <FakeText>- -- -- - --</FakeText>
            <FakeText>- -- ---- --</FakeText>
            <FakeText>- - - --- --</FakeText>
          </>
        ) : (
          <FormationArea>
            <SeabedCloseButton
              onClick={() => onClickClose('formationSection')}
              type="button"
              aria-label={globalTexts.close}
            >
              X
            </SeabedCloseButton>
            <VerticalBar />
            <FormationWrapper>
              <Title>{texts.formation}</Title>
              {formation.sort(customOrder).map(
                (item, index) =>
                  item.visible && (
                    <FormationItem visible key={item.dateBeginValue}>
                      <HorizontalBar
                        border="4px"
                        margin="9%"
                        moveY="25px"
                        width={`${(60 / formation.length) *
                          ((index + 1) * 10)}px`}
                      />
                      <DateArea>
                        {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                        <TextDate>{item.dateBegin}</TextDate>
                        {item.dateEnd && (
                          <>
                            <TextDate>-</TextDate>
                            <TextDate>{item.dateEnd}</TextDate>
                          </>
                        )}
                      </DateArea>
                      <DetailsArea
                        margin={`${(60 / formation.length) *
                          ((index + 1) * 6)}px`}
                      >
                        <FormationTitle>
                          {item.title[languageSelected]}
                        </FormationTitle>
                        <Details>
                          {item.university}
                          <CountryFlag
                            src={getFlagURL(item.countryCode, 'flat', 'small')}
                            alt={item.country[languageSelected]}
                          />
                        </Details>
                        {item.gradeText && (
                          <Details>
                            {item.gradeText[languageSelected]}
                            {item.grade.toFixed(2)}
                          </Details>
                        )}
                      </DetailsArea>
                    </FormationItem>
                  )
              )}
            </FormationWrapper>
          </FormationArea>
        )}
      </SectionFormation>
    </>
  );
};

Formation.propTypes = {
  onClickClose: func.isRequired,
  onClickOpen: func.isRequired,
  status: seabedElementsPropType.isRequired
};

export default Formation;
