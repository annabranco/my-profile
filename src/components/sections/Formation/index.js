import React from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import {
  formationSelector,
  globalTextsSelector,
  currentLanguageSelector,
  seabedTextsSelector
} from '../../../redux/selectors';
import { isDesktop } from '../../../utils/device';
import { getFlagURL } from '../../../utils/icons';
import { seabedElementsPropType } from '../../../types/propTypes';
import { HorizontalBar } from '../../elements/HorizontalBar/styles';
import { Title } from '../../mainComponents/ScrollArea/ScrollSection/styles';
import { CountryFlag, DateArea, TextDate } from '../Experiences/styles';
import { TextFindSomething } from '../Seabed/styles';
import {
  Details,
  DetailsArea,
  FakeText,
  FormationArea,
  FormationItem,
  FormationTitle,
  FormationWrapper,
  SectionFormation,
  VerticalBar
} from './styles';

const FORMATION_ON_TOP = 'older'; // newer or older

const Formation = ({
  onClickClose,
  onClickOpen,
  status: { read, visible }
}) => {
  const formation = useSelector(formationSelector);
  const languageSelected = useSelector(currentLanguageSelector);
  const texts = useSelector(seabedTextsSelector);
  const globalTexts = useSelector(globalTextsSelector);

  const customOrder = (x, y) => {
    if (FORMATION_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <>
      {!read && isDesktop ? (
        <>
          <TextFindSomething>{texts.find}</TextFindSomething>
          <TextFindSomething>{texts.investigate}</TextFindSomething>
        </>
      ) : null}
      <SectionFormation
        id="Formation Section"
        onClick={() => onClickOpen('formationSection')}
        role="button"
        aria-label={globalTexts.open}
        tabIndex={0}
        visible={visible}
      >
        {!visible ? (
          <>
            <FakeText>- - --- - --</FakeText>
            <FakeText>- -- -- - --</FakeText>
            <FakeText>- -- ---- --</FakeText>
            <FakeText>- - - --- --</FakeText>
          </>
        ) : (
          <FormationArea onClick={() => onClickClose('formationSection')}>
            <VerticalBar />
            <FormationWrapper>
              {isDesktop && <Title>{texts.formation}</Title>}
              {formation.sort(customOrder).map(
                (item, index) =>
                  item.visible && (
                    <FormationItem visible key={item.dateBeginValue}>
                      <HorizontalBar
                        border="4px"
                        formationItems={formation.length}
                        index={index}
                        margin="9%"
                        moveY="25px"
                      />
                      <DateArea>
                        {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                        <TextDate>{item.dateBegin}</TextDate>
                        {item.dateEnd && (
                          <>
                            <TextDate>-</TextDate>
                            <TextDate>{item.dateEnd}</TextDate>
                          </>
                        )}
                      </DateArea>
                      <DetailsArea
                        formationItems={formation.length}
                        index={index}
                      >
                        <FormationTitle>
                          {item.title[languageSelected]}
                        </FormationTitle>
                        <Details>
                          {item.university}
                          <CountryFlag
                            alt={item.country[languageSelected]}
                            formation
                            round
                            src={getFlagURL({
                              country: item.countryCode,
                              size: 'small',
                              style: 'round3d'
                            })}
                          />
                        </Details>
                        {item.gradeText && (
                          <Details subtext>
                            {item.gradeText[languageSelected]}
                            {item.grade.toFixed(2)}
                          </Details>
                        )}
                      </DetailsArea>
                    </FormationItem>
                  )
              )}
            </FormationWrapper>
          </FormationArea>
        )}
      </SectionFormation>
    </>
  );
};

Formation.propTypes = {
  onClickClose: func,
  onClickOpen: func,
  status: seabedElementsPropType.isRequired
};

Formation.defaultProps = {
  onClickClose: () => null,
  onClickOpen: () => null
};

export default Formation;
