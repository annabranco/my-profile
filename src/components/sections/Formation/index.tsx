import React, { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { func } from 'prop-types';
import {
  IFormation,
  IGlobalTexts,
  LanguageCode,
  ISeabedElements,
  ISeabedText
} from '../../../types/interfaces';
import {
  currentLanguageSelector,
  formationSelector,
  globalTextsSelector,
  seabedTextsSelector
} from '../../../redux/selectors';
import { isDesktop } from '../../../utils/device';
import { getFlagURL } from '../../../utils/icons';
import { seabedElementsPropType } from '../../../types/propTypes';
import { HorizontalBar } from '../../elements/HorizontalBar/styles';
import { Title } from '../../mainComponents/ScrollArea/ScrollSection/styles';
import { CountryFlag, DateArea, TextDate } from '../Experiences/styles';
import { TextFindSomething } from '../Seabed/styles';
import {
  Details,
  DetailsArea,
  FakeText,
  FormationArea,
  FormationItem,
  FormationTitle,
  FormationWrapper,
  SectionFormation,
  VerticalBar
} from './styles';

const FORMATION_ON_TOP = 'older'; // newer or older

interface Props {
  onClickClose: (type: string) => void;
  onClickOpen: (type: string) => void;
  status: ISeabedElements;
}

const Formation = ({
  onClickClose,
  onClickOpen,
  status: { read, visible }
}: Props): ReactElement => {
  const formation: IFormation[] = useSelector(formationSelector);
  const languageSelected: LanguageCode = useSelector(currentLanguageSelector);
  const texts: ISeabedText = useSelector(seabedTextsSelector);
  const globalTexts: IGlobalTexts = useSelector(globalTextsSelector);

  const customOrder = (x: IFormation, y: IFormation) => {
    if (FORMATION_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <>
      {!read && isDesktop ? (
        <>
          <TextFindSomething>{texts.find}</TextFindSomething>
          <TextFindSomething>{texts.investigate}</TextFindSomething>
        </>
      ) : null}
      <SectionFormation
        id="Formation Section"
        onClick={() => onClickOpen('formationSection')}
        role="button"
        aria-label={globalTexts.open}
        tabIndex={0}
        visible={visible}
      >
        {!visible ? (
          <>
            <FakeText>- - --- - --</FakeText>
            <FakeText>- -- -- - --</FakeText>
            <FakeText>- -- ---- --</FakeText>
            <FakeText>- - - --- --</FakeText>
          </>
        ) : (
          <FormationArea onClick={() => onClickClose('formationSection')}>
            <VerticalBar />
            <FormationWrapper>
              {isDesktop && <Title>{texts.formation}</Title>}
              {formation.sort(customOrder).map(
                (item, index) =>
                  item.visible && (
                    <FormationItem key={item.dateBeginValue}>
                      <HorizontalBar
                        border="4px"
                        formationItems={formation.length}
                        index={index}
                        margin="9%"
                        moveY="25px"
                      />
                      <DateArea>
                        {!item.dateEnd && (
                          <TextDate>{globalTexts.since}</TextDate>
                        )}
                        <TextDate>{item.dateBegin}</TextDate>
                        {item.dateEnd && (
                          <>
                            <TextDate>-</TextDate>
                            <TextDate>{item.dateEnd}</TextDate>
                          </>
                        )}
                      </DateArea>
                      <DetailsArea
                        formationItems={formation.length}
                        index={index}
                      >
                        <FormationTitle>
                          {item.title[languageSelected]}
                        </FormationTitle>
                        <Details>
                          {item.university}
                          <CountryFlag
                            alt={item.country[languageSelected]}
                            formation
                            round
                            src={getFlagURL({
                              country: item.country.countryCode,
                              size: 'small',
                              style: 'round3d'
                            })}
                          />
                        </Details>
                        {item.gradeText && (
                          <Details subtext>
                            {item.gradeText[languageSelected]}
                            {item.grade.toFixed(2)}
                          </Details>
                        )}
                      </DetailsArea>
                    </FormationItem>
                  )
              )}
            </FormationWrapper>
          </FormationArea>
        )}
      </SectionFormation>
    </>
  );
};

Formation.propTypes = {
  onClickClose: func,
  onClickOpen: func,
  status: seabedElementsPropType.isRequired
};

Formation.defaultProps = {
  onClickClose: () => null,
  onClickOpen: () => null
};

export default Formation;
