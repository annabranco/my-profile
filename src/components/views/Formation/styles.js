import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlueDark,
  colorBlack,
  fontSubtitle,
  fontTitleAlt,
  fontTitle,
  colorYellowBright,
  colorBlueWater,
  colorRedDark,
  colorYellowDark
} from '../../../styles/theme';
import { Logo } from '../Skills/styles';

export const SectionFormation = styled.section`
  z-index: 6;
  position: absolute;
  top: initial;
  bottom: 50px;
  left: 20%;
  opacity: 0.5;
  transform: perspective(5em) rotateX(50deg);
  outline: none;
  border: 1px solid ${colorBlack};
  border-radius: 10px 0;
  background-image: linear-gradient(
    to top right,
    ${colorWhite},
    ${colorBlueWater},
    ${colorBlueLight}
  );
  height: 40px;
  width: 60px;
  overflow: hidden;
  cursor: pointer;
  box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 768px) {
    left: 40%;
  }

  &:hover {
    opacity: 0.7;
    border: 3px solid ${colorYellowBright};
  }

  ${({ visible }) =>
    visible &&
    css`
      bottom: 0;
      left: 0;
      opacity: 0.95;
      transform: none;
      border: 2px solid ${colorBlack};
      border-radius: 0;
      height: 90vh;
      width: 100%;
      padding: 5px;
      cursor: default;

      &:hover {
        border: 2px solid ${colorBlack};
        opacity: 0.95;
      }

      @media all and (min-width: 768px) {
        top: 40px;
        bottom: initial;
        left: 20vw;
        opacity: 0.95;
        transform: none;
        border: 2px solid ${colorBlack};
        border-radius: 40px 0;
        height: 90vh;
        width: 75vw;
        padding: 15px;
      }
    `}
`;
SectionFormation.displayName = '--- FORMATION Section ---';

export const FakeText = styled.p`
  width: 100%;
  font-size: 0.9rem;
  line-height: 0.8;
  text-align: center;
`;
FakeText.displayName = '--- Fake Text ---';

export const FormationArea = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 0 auto;
  height: 100%;
  width: 100%;
`;
FormationArea.displayName = '--- Formation Area ---';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = '--- Vertical Bar ---';

export const FormationWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-end;
  margin: 0 auto;
  padding-right: 5vw;
  height: auto;
  width: 100%;
`;
FormationWrapper.displayName = '--- Formation Wrapper ---';

export const FormationItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 20px 0;
  height: auto;
  width: 100%;
`;
FormationItem.displayName = '--- Formation Item ---';

export const FormationTitle = styled.h2`
  display: inline;
  margin-left: 20px;
  font-family: ${fontTitleAlt};
  font-size: 1rem;
  color: ${colorBlueDark};

  @media all and (min-width: 321px) {
    font-size: 1.4rem;
  }

  &:hover {
    letter-spacing: 0.1rem;
    color: ${colorYellowDark};
  }
`;
FormationTitle.displayName = '--- Formation Title ---';

export const DetailsArea = styled.div`
  width: 80%;

  ${({ margin }) =>
    margin &&
    css`
      margin-left: ${margin};
    `}
`;
DetailsArea.displayName = '--- Details Area ---';

export const Details = styled.p`
  margin: 0 40px;
  font-family: ${fontSubtitle};
  font-size: 0.9rem;
  line-height: 1;

  @media all and (min-width: 768px) {
    font-size: 1rem;
    line-height: 1.3;
  }
  &:first-of-type {
    margin-top: 5px;
  }
`;
Details.displayName = '--- Details ---';

// export const Flag = styled.img`
//   margin-left: 5px;
//   vertical-align: middle;
// `;
// Flag.displayName = '--- Flag ---';

// export const MoreInfo = styled.p`
//   position: absolute;
//   bottom: 5px;
//   right: 35px;
//   font-size: 0.8rem;

//   @media all and (min-width: 321px) {
//     bottom: 15px;
//     font-size: 1rem;
//   }
// `;
// MoreInfo.displayName = '--- More Info ---';

// export const Link = styled.a`
//   margin-left: 5px;
//   vertical-align: super;
// `;
// Link.displayName = '--- Link ---';
