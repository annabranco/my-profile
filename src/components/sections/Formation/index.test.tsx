import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import '../../../tests/mocks/matchMedia';
import Formation from '.';
import { setupMockProvider, subSectionsStatus } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< Formation >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <Formation
          onClickClose={() => null}
          onClickOpen={() => null}
          status={subSectionsStatus}
        />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render an event message on desktops', () => {});

  xit('should display a section title', () => {});

  xit('should render a list with formation received from server', () => {});

  xit('should organize the list according the the config order', () => {});

  xit('should display date, formation title, university and country flag of each formation', () => {});

  xit('should render the grade of each formation, if it exists on the data', () => {});

  xit('should translate all texts when the language is changed', () => {});
});
