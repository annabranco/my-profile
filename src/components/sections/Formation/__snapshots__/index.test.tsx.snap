// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Formation > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Formation
    onClickClose={[Function]}
    onClickOpen={[Function]}
    status={
      Object {
        "active": false,
        "read": false,
        "visible": false,
      }
    }
  />
</Provider>
`;
